jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -422,24 +506,38 @@ object XGBoost extends Serializable {
       case "_reg_" =>
         val xgBoostModel = new XGBoostRegressionModel(SXGBoost.loadModel(dataInStream))
         setGeneralModelParams(featureCol, labelCol, predictionCol, xgBoostModel)
+      case other =>
+        throw new XGBoostError(s"Unknown model type $other. Supported types " +
+          s"are: ['_reg_', '_cls_'].")
     }
   }
 }
 
+/** Class to hold the training and test matrices in a given partition. */
```

Comment:
inconsistent comment style

Reply:
hm, not sure what you mean?

Reply:
for single line comment, we use `//`, additionally, do we really need such a line to explain `class Watches private(val train: DMatrix, val test: DMatrix)` means it is a class containing two matrices ?

Reply:
it's a concept in the context of XGBoost which appears even earlier than when I was involved in the project

https://github.com/dmlc/xgboost/blob/72451457120ac9d59573cf7580ccd2ad178ef908/demo/guide-python/basic_walkthrough.py#L16